workspaces/local-cli/src/shared/local-cli-task-runner.ts

```diff
@@ -167,7 +199,29 @@ ${blockers.map((x) => `[pid ${x.pid}]: ${x.cmd}`).join('\n')}
 
     ////////////////////////////////////////////////////////////////////////////////
     process.env.OPTIC_ENABLE_CAPTURE_BODY = 'yes';
-    const sessionManager = new CommandAndProxySessionManager(taskConfig);
+
+    const testCommand = commandToRunWhenStarted
+      ? async () => {
+          console.log(
+            fromOptic(
+              'Running test command ' +
+                colors.grey.bold(commandToRunWhenStarted)
+            )
+          );
+
+          await spawnProcess(commandToRunWhenStarted!, {
+            OPTIC_PROXY_PORT: taskConfig.proxyConfig.port.toString(),
+            OPTIC_PROXY_HOST: taskConfig.proxyConfig.host.toString(),
+            OPTIC_PROXY: `http://${taskConfig.proxyConfig.host.toString()}:${taskConfig.proxyConfig.port.toString()}`,
+          });
+          return new Promise((resolve) => setTimeout(resolve, 500)); // needs time to finish saving
```

Comment:
command-and-proxy-session-manager.ts line 100 calls persistenceManager.cleanup(), which in this flow is the file-system/capture-saver-with-diffs.ts, which extends capture-saver.ts. the saas implementation handles the asynchronous calls to save() that batch after cleanup() is called. So, a less time-based and more event-based way would be to modify cleanup() so it stops receiving and waits for the batches to finish before resolving cleanup()

Reply:
it's not obvious to me what I should await in that cleanup -- where's the example for agents?

Reply:
workspaces/cli-shared/src/command-and-proxy-session-manager.ts:34 is where we save interactions. We never wait for those promises, so can add an implicit guarantee that cleanup() ensures interactions have been flushed, or perhaps better and more simply, the command-and-proxy-session-manager can add those promises to a structure like a task queue (e.g. using bottleneck) and wait for the queue to be empty when stopping

Reply:
So you probably won't end up messing with cleanup, but here's what the agents use: workspaces/cli-shared/src/captures/avro/saas/capture-saver.ts

Reply:
@devdoshi check out the latest. I'm saving the promises and doing `promise.all()`. this might be memory unfriendly at high volumes though.